pkg/controller/kibana/pod.go

```diff
@@ -137,9 +140,9 @@ func NewPodTemplateSpec(
 	// Limiting to 7.10.0 here as there was a bug in previous versions causing rebuilding
 	// of browser bundles to happen on plugin install, which would attempt a write to the
 	// root filesystem on restart.
-	if v.GTE(version.From(7, 10, 0)) {
-		builder.WithPodSecurityContext(defaultPodSecurityContext).
-			WithContainersSecurityContext(defaultSecurityContext).
+	if v.GTE(version.From(7, 10, 0)) && setDefaultSecurityContext {
```

Comment:
A bit late but I think this is not consistent with Elasticsearch for which only the addition of the [Pod SecurityContext](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#podsecuritycontext-v1-core) is conditioned by this variable:
https://github.com/elastic/cloud-on-k8s/blob/59aada97b49774a9ac0de712822067525973c6ad/pkg/controller/elasticsearch/nodespec/podspec.go#L91-L94

While we always set a [SecurityContext for the containers](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#securitycontext-v1-core) to set things like `readOnlyRootFilesystem` or `allowPrivilegeEscalation`:
https://github.com/elastic/cloud-on-k8s/blob/59aada97b49774a9ac0de712822067525973c6ad/pkg/controller/elasticsearch/nodespec/podspec.go#L134

Not a blocker for `2.16` but we may want to make it consistent.